### PR TITLE
Port Restriction Policy 2

### PR DESCRIPTION
**Pull request by markandrus**
@njohnst I took #368 and ran with the `min`/`max` idea, building on work from #370. I still need to do some more testing, but ExtendedRTCConfiguration should give us a pattern for adding future custom RTCConfiguration options if we deem them useful. **EDIT:** Also, there are some lint errors to address in the custom-settings.js test.

### CODE REVIEW

File: src/peerconnection.cc [hunk 1]
```diff
@@ -64,7 +65,20 @@ PeerConnection::PeerConnection(RTCConfiguration configuration)
...
   _shouldReleaseFactory = true;
 
-  _jinglePeerConnection = _factory->factory()->CreatePeerConnection(configuration, nullptr, nullptr, nullptr, this);
+  auto portAllocator = std::unique_ptr<cricket::PortAllocator>(new cricket::BasicPortAllocator(
+              _factory->getNetworkManager(),
+              _factory->getSocketFactory()
+          ));
+  portAllocator->SetPortRange(
+      configuration.portRange.min.FromMaybe(0),
+      configuration.portRange.max.FromMaybe(65535));
```
Comment (njohnst) [inline src/peerconnection.cc]: bracket on separate line ?

**Comment by njohnst** [comment]
Looks great, a lot cleaner to deal with the config parsing this way
Thank you for fixing the lint errors as well
In peerconnection.cc I spotted a few places where the parentheses/line breaks were slightly inconsistent (lines 68-81), not sure if that's worth fixing

File: src/peerconnection.cc (hunk 1 above)
Comment (markandrus) [reply src/peerconnection.cc]: Ah, good catch. I moved the bracket to the same line instead.

**Comment by markandrus** [comment]
Thanks, @njohnst!

**Pull request merged**